### PR TITLE
Core: Replace Action class uses with GenericAction

### PR DESCRIPTION
**Pull request by rjernst**
Since #30966, Action no longer has anything but a call to the
GenericAction super constructor. This commit replaces all uses of Action
with GenericAction, and removes Action. It also removes Request as a
generic type of GenericAction, as it was unused.

### CODE REVIEW

**Review by bleskes — APPROVED** [review]
LGTM.

File: modules/reindex/src/test/java/org/elasticsearch/index/reindex/ReindexFromRemoteWithAuthTests.java [hunk 1]
```diff
@@ -166,7 +166,7 @@ public void testReindexWithBadAuthentication() throws Exception {
     }
 
     /**
-     * Action filter that will reject the request if it isn't authenticated.
+     * GenericAction filter that will reject the request if it isn't authenticated.
```
Comment (nik9000) [inline modules/reindex/src/test/java/org/elasticsearch/index/reindex/ReindexFromRemoteWithAuthTests.java]: Is this change intentional?

File: server/src/main/java/org/elasticsearch/action/ActionModule.java [hunk 2]
```diff
@@ -405,8 +405,8 @@ public void register(ActionHandler<?, ?> handler) {
             }
 
             public <Request extends ActionRequest, Response extends ActionResponse> void register(
-                    GenericAction<Request, Response> action, Class<? extends TransportAction<Request, Response>> transportAction,
-                    Class<?>... supportTransportActions) {
+                GenericAction<Response> action, Class<? extends TransportAction<Request, Response>> transportAction,
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/action/ActionModule.java]: Your indentation change here makes this harder to read.

File: server/src/main/java/org/elasticsearch/action/DocWriteRequest.java [hunk 3]
```diff
@@ -31,7 +31,7 @@
 
 /**
  * Generic interface to group ActionRequest, which perform writes to a single document
- * Action requests implementing this can be part of {@link org.elasticsearch.action.bulk.BulkRequest}
+ * GenericAction requests implementing this can be part of {@link org.elasticsearch.action.bulk.BulkRequest}
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/action/DocWriteRequest.java]: I think this was right before?

File: server/src/main/java/org/elasticsearch/action/bulk/BulkRequest.java [hunk 4]
```diff
@@ -369,7 +369,7 @@ public BulkRequest add(BytesReference data, @Nullable String defaultIndex, @Null
                             } else if (SOURCE.match(currentFieldName, parser.getDeprecationHandler())) {
                                 fetchSourceContext = FetchSourceContext.fromXContent(parser);
                             } else {
-                                throw new IllegalArgumentException("Action/metadata line [" + line + "] contains an unknown parameter [" + currentFieldName + "]");
+                                throw new IllegalArgumentException("GenericAction/metadata line [" + line + "] contains an unknown parameter [" + currentFieldName + "]");
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/action/bulk/BulkRequest.java]: I think this was right before? Or, at least, this change doesn't make it more right.

File: server/src/main/java/org/elasticsearch/action/support/ThreadedActionListener.java [hunk 5]
```diff
@@ -48,7 +48,7 @@
         public Wrapper(Logger logger, Settings settings, ThreadPool threadPool) {
             this.logger = logger;
             this.threadPool = threadPool;
-             // Should the action listener be threaded or not by default. Action listeners are automatically threaded for
+             // Should the action listener be threaded or not by default. GenericAction listeners are automatically threaded for
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/action/support/ThreadedActionListener.java]: Same here. I think they are still called action listeners.

File: server/src/main/java/org/elasticsearch/client/transport/TransportProxyClient.java [hunk 6]
```diff
@@ -38,22 +37,22 @@
...
 
     TransportProxyClient(Settings settings, TransportService transportService,
                                 TransportClientNodesService nodesService, List<GenericAction> actions) {
         this.nodesService = nodesService;
-        Map<Action, TransportActionNodeProxy> proxies = new HashMap<>();
+        Map<GenericAction, TransportActionNodeProxy> proxies = new HashMap<>();
         for (GenericAction action : actions) {
-            if (action instanceof Action) {
-                proxies.put((Action) action, new TransportActionNodeProxy(settings, action, transportService));
+            if (action instanceof GenericAction) {
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/client/transport/TransportProxyClient.java]: I thinkn this `instanceof` is always true.

File: server/src/main/java/org/elasticsearch/cluster/service/TaskBatcher.java [hunk 7]
```diff
@@ -115,7 +115,7 @@ private void onTimeoutInternal(List<? extends BatchedTask> tasks, TimeValue time
     }
 
     /**
-     * Action to be implemented by the specific batching implementation.
+     * GenericAction to be implemented by the specific batching implementation.
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/cluster/service/TaskBatcher.java]: I think the old text was better

File: server/src/main/java/org/elasticsearch/cluster/service/TaskBatcher.java [hunk 8]
```diff
@@ -153,7 +153,7 @@ void runIfNotProcessed(BatchedTask updateTask) {
     }
 
     /**
-     * Action to be implemented by the specific batching implementation
+     * GenericAction to be implemented by the specific batching implementation
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/cluster/service/TaskBatcher.java]: I think the old text was better.

File: server/src/main/java/org/elasticsearch/persistent/PersistentTasksNodeService.java [hunk 9]
```diff
@@ -78,7 +78,7 @@ public void clusterChanged(ClusterChangedEvent event) {
         PersistentTasksCustomMetaData tasks = event.state().getMetaData().custom(PersistentTasksCustomMetaData.TYPE);
         PersistentTasksCustomMetaData previousTasks = event.previousState().getMetaData().custom(PersistentTasksCustomMetaData.TYPE);
 
-        // Cluster State   Local State      Local Action
+        // Cluster State   Local State      Local GenericAction
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/persistent/PersistentTasksNodeService.java]: I'm not sure about this change either.

File: server/src/main/java/org/elasticsearch/plugins/ActionPlugin.java [hunk 10]
```diff
@@ -73,7 +73,7 @@
     }
 
     /**
-     * Action filters added by this plugin.
+     * GenericAction filters added by this plugin.
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/plugins/ActionPlugin.java]: They are still called action filters.

File: server/src/main/java/org/elasticsearch/transport/TransportService.java [hunk 11]
```diff
@@ -654,7 +654,7 @@ private void sendLocalRequest(long requestId, final String action, final Transpo
             onRequestReceived(requestId, action);
             final RequestHandlerRegistry reg = getRequestHandler(action);
             if (reg == null) {
-                throw new ActionNotFoundTransportException("Action [" + action + "] not found");
+                throw new ActionNotFoundTransportException("GenericAction [" + action + "] not found");
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/transport/TransportService.java]: I think the old message was fine.

File: server/src/test/java/org/elasticsearch/action/admin/cluster/node/tasks/TransportTasksActionTests.java [hunk 12]
```diff
@@ -300,14 +300,14 @@ private Task startBlockingTestNodesAction(CountDownLatch checkLatch, NodesReques
...
-                    logger.info("Action on node {}", node);
+                    logger.info("GenericAction on node {}", node);
                     actionLatch.countDown();
                     try {
                         checkLatch.await();
                     } catch (InterruptedException ex) {
                         Thread.currentThread().interrupt();
                     }
-                    logger.info("Action on node {} finished", node);
+                    logger.info("GenericAction on node {} finished", node);
```
Comment (nik9000) [inline server/src/test/java/org/elasticsearch/action/admin/cluster/node/tasks/TransportTasksActionTests.java]: I think the old message was fine.

File: server/src/test/java/org/elasticsearch/action/admin/cluster/node/tasks/TransportTasksActionTests.java [hunk 13]
```diff
@@ -585,7 +585,7 @@ public void testFailedTasksCount() throws ExecutionException, InterruptedExcepti
                     testNodes[i].transportService) {
                 @Override
                 protected NodeResponse nodeOperation(NodeRequest request) {
-                    logger.info("Action on node {}", node);
+                    logger.info("GenericAction on node {}", node);
```
Comment (nik9000) [inline server/src/test/java/org/elasticsearch/action/admin/cluster/node/tasks/TransportTasksActionTests.java]: I think the old message was fine.

File: server/src/test/java/org/elasticsearch/persistent/PersistentTasksNodeServiceTests.java [hunk 14]
```diff
@@ -139,7 +139,7 @@ public void testStartTask() {
 
         coordinator.clusterChanged(new ClusterChangedEvent("test", newClusterState, state));
         if (added) {
-            // Action for this node was added, let's make sure it was invoked
+            // GenericAction for this node was added, let's make sure it was invoked
```
Comment (nik9000) [inline server/src/test/java/org/elasticsearch/persistent/PersistentTasksNodeServiceTests.java]: I think the old comment was fine.

File: test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java [hunk 15]
```diff
@@ -1613,7 +1613,7 @@ public final void onResponse(Response response) {
         @Override
         public final void onFailure(Exception t) {
             try {
-                logger.info("Action Failed", t);
+                logger.info("GenericAction Failed", t);
```
Comment (nik9000) [inline test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java]: I think the old log was fine.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ClientHelper.java [hunk 16]
```diff
@@ -81,8 +81,8 @@ public static Client clientWithOrigin(Client client, String origin) {
      */
     public static <Request extends ActionRequest, Response extends ActionResponse,
             RequestBuilder extends ActionRequestBuilder<Request, Response>> void executeAsyncWithOrigin(
-            Client client, String origin, Action<Request, Response> action, Request request,
-            ActionListener<Response> listener) {
+        Client client, String origin, GenericAction<Response> action, Request request,
```
Comment (nik9000) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ClientHelper.java]: The old indentation was easier to read.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/RevertModelSnapshotAction.java [hunk 17]
```diff
@@ -30,7 +30,7 @@
 import java.util.Objects;
 
 public class RevertModelSnapshotAction
-extends Action<RevertModelSnapshotAction.Request, RevertModelSnapshotAction.Response> {
+extends GenericAction<RevertModelSnapshotAction.Response> {
```
Comment (nik9000) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/RevertModelSnapshotAction.java]: Maybe we don't need the new line here any more. That'd be nice!

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/role/PutRoleAction.java [hunk 18]
```diff
@@ -5,12 +5,12 @@
  */
 package org.elasticsearch.xpack.core.security.action.role;
 
-import org.elasticsearch.action.Action;
+import org.elasticsearch.action.GenericAction;
 
 /**
- * Action for adding a role to the security index
+ * GenericAction for adding a role to the security index
```
Comment (nik9000) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/role/PutRoleAction.java]: I think the old comment was fine.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/role/GetRolesAction.java [hunk 19]
```diff
@@ -5,12 +5,12 @@
  */
 package org.elasticsearch.xpack.core.security.action.role;
 
-import org.elasticsearch.action.Action;
+import org.elasticsearch.action.GenericAction;
 
 /**
- * Action to retrieve a role from the security index
+ * GenericAction to retrieve a role from the security index
```
Comment (nik9000) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/role/GetRolesAction.java]: I think the old comment was fine.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/role/DeleteRoleAction.java [hunk 20]
```diff
@@ -5,12 +5,12 @@
  */
 package org.elasticsearch.xpack.core.security.action.role;
 
-import org.elasticsearch.action.Action;
+import org.elasticsearch.action.GenericAction;
 
 /**
- * Action for deleting a role from the security index
+ * GenericAction for deleting a role from the security index
```
Comment (nik9000) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/role/DeleteRoleAction.java]: I think the old comment was fine.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/rolemapping/GetRoleMappingsAction.java [hunk 21]
```diff
@@ -5,14 +5,14 @@
  */
 package org.elasticsearch.xpack.core.security.action.rolemapping;
 
-import org.elasticsearch.action.Action;
+import org.elasticsearch.action.GenericAction;
 
 /**
- * Action to retrieve one or more role-mappings from X-Pack security
+ * GenericAction to retrieve one or more role-mappings from X-Pack security
```
Comment (nik9000) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/rolemapping/GetRoleMappingsAction.java]: I think the old comment was fine.

Can you remove the extra new line below this line while you are here?

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/saml/SamlAuthenticateAction.java [hunk 22]
```diff
@@ -5,13 +5,12 @@
  */
 package org.elasticsearch.xpack.core.security.action.saml;
 
-import org.elasticsearch.action.Action;
-import org.elasticsearch.client.ElasticsearchClient;
+import org.elasticsearch.action.GenericAction;
 
 /**
- * Action for authenticating using SAML assertions
+ * GenericAction for authenticating using SAML assertions
```
Comment (nik9000) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/saml/SamlAuthenticateAction.java]: I think the old comment was fine.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/rolemapping/PutRoleMappingAction.java [hunk 23]
```diff
@@ -5,13 +5,12 @@
  */
 package org.elasticsearch.xpack.core.security.action.rolemapping;
 
-import org.elasticsearch.action.Action;
-import org.elasticsearch.client.ElasticsearchClient;
+import org.elasticsearch.action.GenericAction;
 
 /**
- * Action for adding a role to the security index
+ * GenericAction for adding a role to the security index
```
Comment (nik9000) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/rolemapping/PutRoleMappingAction.java]: I think the old comment was fine.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/saml/SamlInvalidateSessionAction.java [hunk 24]
```diff
@@ -5,13 +5,12 @@
  */
 package org.elasticsearch.xpack.core.security.action.saml;
 
-import org.elasticsearch.action.Action;
-import org.elasticsearch.client.ElasticsearchClient;
+import org.elasticsearch.action.GenericAction;
 
 /**
- * Action to perform IdP-initiated logout for a SAML-SSO user
+ * GenericAction to perform IdP-initiated logout for a SAML-SSO user
```
Comment (nik9000) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/saml/SamlInvalidateSessionAction.java]: I think the old comment was fine.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/saml/SamlLogoutAction.java [hunk 25]
```diff
@@ -5,12 +5,12 @@
  */
 package org.elasticsearch.xpack.core.security.action.saml;
 
-import org.elasticsearch.action.Action;
+import org.elasticsearch.action.GenericAction;
 
 /**
- * Action for initiating a logout process for a SAML-SSO user
+ * GenericAction for initiating a logout process for a SAML-SSO user
```
Comment (nik9000) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/saml/SamlLogoutAction.java]: I think the old comment was fine.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/saml/SamlPrepareAuthenticationAction.java [hunk 26]
```diff
@@ -5,13 +5,12 @@
  */
 package org.elasticsearch.xpack.core.security.action.saml;
 
-import org.elasticsearch.action.Action;
-import org.elasticsearch.client.ElasticsearchClient;
+import org.elasticsearch.action.GenericAction;
 
 /**
- * Action for initiating an authentication process using SAML assertions
+ * GenericAction for initiating an authentication process using SAML assertions
```
Comment (nik9000) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/saml/SamlPrepareAuthenticationAction.java]: I think the old comment was fine.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/token/CreateTokenAction.java [hunk 27]
```diff
@@ -5,12 +5,12 @@
  */
 package org.elasticsearch.xpack.core.security.action.token;
 
-import org.elasticsearch.action.Action;
+import org.elasticsearch.action.GenericAction;
 
 /**
- * Action for creating a new token
+ * GenericAction for creating a new token
```
Comment (nik9000) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/token/CreateTokenAction.java]: I think the old comment was fine.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/token/InvalidateTokenAction.java [hunk 28]
```diff
@@ -5,12 +5,12 @@
  */
 package org.elasticsearch.xpack.core.security.action.token;
 
-import org.elasticsearch.action.Action;
+import org.elasticsearch.action.GenericAction;
 
 /**
- * Action for invalidating a given token
+ * GenericAction for invalidating a given token
```
Comment (nik9000) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/token/InvalidateTokenAction.java]: I think the old comment was fine.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/user/DeleteUserAction.java [hunk 29]
```diff
@@ -5,12 +5,12 @@
  */
 package org.elasticsearch.xpack.core.security.action.user;
 
-import org.elasticsearch.action.Action;
+import org.elasticsearch.action.GenericAction;
 
 /**
- * Action for deleting a native user.
+ * GenericAction for deleting a native user.
```
Comment (nik9000) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/user/DeleteUserAction.java]: I think the old comment was fine.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/user/GetUsersAction.java [hunk 30]
```diff
@@ -5,12 +5,12 @@
  */
 package org.elasticsearch.xpack.core.security.action.user;
 
-import org.elasticsearch.action.Action;
+import org.elasticsearch.action.GenericAction;
 
 /**
- * Action for retrieving a user from the security index
+ * GenericAction for retrieving a user from the security index
```
Comment (nik9000) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/user/GetUsersAction.java]: I think the old comment was fine.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/user/PutUserAction.java [hunk 31]
```diff
@@ -5,12 +5,12 @@
  */
 package org.elasticsearch.xpack.core.security.action.user;
 
-import org.elasticsearch.action.Action;
+import org.elasticsearch.action.GenericAction;
 
 /**
- * Action for putting (adding/updating) a native user.
+ * GenericAction for putting (adding/updating) a native user.
```
Comment (nik9000) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/user/PutUserAction.java]: I think the old comment was fine.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/action/GetCertificateInfoAction.java [hunk 32]
```diff
@@ -22,10 +22,10 @@
 import java.util.Collection;
 
 /**
- * Action to obtain information about X.509 (SSL/TLS) certificates that are being used by X-Pack.
+ * GenericAction to obtain information about X.509 (SSL/TLS) certificates that are being used by X-Pack.
```
Comment (nik9000) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/action/GetCertificateInfoAction.java]: I think the old comment was fine.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/watcher/actions/ActionWrapper.java [hunk 33]
```diff
@@ -83,7 +83,7 @@ public ExecutableAction action() {
      * <li>Throttling</li>
      * <li>Conditional Check</li>
      * <li>Transformation</li>
-     * <li>Action</li>
+     * <li>GenericAction</li>
```
Comment (nik9000) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/watcher/actions/ActionWrapper.java]: I think the old comment was fine.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/watcher/transport/actions/stats/WatcherStatsAction.java [hunk 34]
```diff
@@ -5,12 +5,12 @@
  */
 package org.elasticsearch.xpack.core.watcher.transport.actions.stats;
 
-import org.elasticsearch.action.Action;
+import org.elasticsearch.action.GenericAction;
 
 /**
- * This Action gets the stats for the watcher plugin
+ * This GenericAction gets the stats for the watcher plugin
```
Comment (nik9000) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/watcher/transport/actions/stats/WatcherStatsAction.java]: I think the old comment was fine.

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobDataCountsPersister.java [hunk 35]
```diff
@@ -49,7 +49,7 @@ private XContentBuilder serialiseCounts(DataCounts counts) throws IOException {
      *
      * @param jobId Job to update
      * @param counts The counts
-     * @param listener Action response listener
+     * @param listener GenericAction response listener
```
Comment (nik9000) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobDataCountsPersister.java]: I think the old comment was better.

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java [hunk 36]
```diff
@@ -328,7 +328,7 @@ public void authorize(Authentication authentication, String action, TransportReq
         if (action.equals(TransportShardBulkAction.ACTION_NAME)) {
             // is this is performing multiple actions on the index, then check each of those actions.
             assert request instanceof BulkShardRequest
-                    : "Action " + action + " requires " + BulkShardRequest.class + " but was " + request.getClass();
+                    : "GenericAction " + action + " requires " + BulkShardRequest.class + " but was " + request.getClass();
```
Comment (nik9000) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java]: I think the old message was fine,.

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java [hunk 37]
```diff
@@ -523,7 +523,7 @@ static boolean checkSameUserPermissions(String action, TransportRequest request,
             }
 
             assert AuthenticateAction.NAME.equals(action) || HasPrivilegesAction.NAME.equals(action) || sameUsername == false
-                    : "Action '" + action + "' should not be possible when sameUsername=" + sameUsername;
+                    : "GenericAction '" + action + "' should not be possible when sameUsername=" + sameUsername;
```
Comment (nik9000) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java]: I think the old message was fine.

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/audit/index/IndexAuditTrailMutedTests.java [hunk 38]
```diff
@@ -72,7 +72,7 @@ public void setup() {
             @Override
             protected <Request extends ActionRequest, Response extends ActionResponse, RequestBuilder extends
                     ActionRequestBuilder<Request, Response>> void doExecute(
-                    Action<Request, Response> action, Request request, ActionListener<Response> listener) {
+                GenericAction<Response> action, Request request, ActionListener<Response> listener) {
```
Comment (nik9000) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/audit/index/IndexAuditTrailMutedTests.java]: The indentation is not longer consistent here.

**Review by nik9000 — COMMENTED** [review]
I wonder if we should rename `GenericAction` to `Action` now.

**Comment by rjernst** [comment]
@nik9000 I opened https://github.com/elastic/elasticsearch/pull/31405 as an alternative to this PR, which does as you suggested (GenericAction -> Action). That also eliminates the IntelliJ wackiness that caused most of the doc changes you saw.

**Comment by nik9000** [comment]
Thanks! I like that one a lot!
